### ISSUE TITLE
react-native init is failing 

### ISSUE THREAD

**Issue by cstayyab** [facebook/react-native#40737]
### Description

New Project creation fails due to https://github.com/CocoaPods/CocoaPods/issues/12081

### React Native Version

0.72.5

### Output of `npx react-native info`

```
info Fetching system and libraries information...
System:
  OS: macOS 14.0
  CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
  Memory: 526.26 MB / 16.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 18.12.1
    path: ~/.nvm/versions/node/v18.12.1/bin/node
  Yarn:
    version: 1.22.19
    path: ~/.nvm/versions/node/v18.12.1/bin/yarn
  npm:
    version: 8.19.2
    path: ~/.nvm/versions/node/v18.12.1/bin/npm
  Watchman:
    version: 2023.10.02.00
    path: /usr/local/bin/watchman
Managers:
  CocoaPods:
    version: 1.13.0
    path: /Users/tayyab/.gem/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.0
      - iOS 17.0
      - macOS 14.0
      - tvOS 17.0
      - watchOS 10.0
  Android SDK: Not Found
IDEs:
  Android Studio: 2022.3 AI-223.8836.35.2231.10811636
  Xcode:
    version: 15.0/15A240d
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 11.0.20.1
    path: /usr/bin/javac
  Ruby:
    version: 3.2.0
    path: /Users/tayyab/.rbenv/shims/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.72.5
    wanted: 0.72.5
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: Not found
  newArchEnabled: Not found
```

### Steps to reproduce

Just create new project using

`npx react-native init test`

### Snack, screenshot, or link to a repository

https://github.com/cstayyab/react-native-issue-40737

<img width="1580" alt="image" src="https://github.com/facebook/react-native/assets/29598866/58c59c9b-fc34-45f1-a752-5ca63e857d28">

**Comment by mamicicekel** [comment]
I'm also getting the same error, and I haven't been able to solve it.

**Comment by AdityaGoyal1999** [comment]
Even I'm getting the same error for the past 3 days. The bundling is not happening properly

**Comment by cstayyab** [comment]
# Workaround:

When `react-native init` fails at cocoapods installation, do the following steps:

1. Go to your project directory and open `Gemfile`
2. Replace Cocoapods version with `1.13`
3. Insert following line at the end of *Gemfile* :  `gem 'activesupport', '~> 7.0.8'`
4. Run `bundle install` in your project directory
5. Go to `ios` directory in your project
6. Run `bundle exec pod install`

Now proceed as you would if `react-native init` was successful

**Comment by mamicicekel** [comment]
> # Workaround:
> When `react-native init` fails at cocoapods installation, do the following steps:
> 
> 1. Go to your project directory and open `Gemfile`
> 2. Replace Cocoapods version with `1.13`
> 3. Insert following line at the end of _Gemfile_ :  `gem 'activesupport', '~> 7.0.8'`
> 4. Run `bundle install` in your project directory
> 5. Go to `ios` directory in your project
> 6. Run `bundle exec pod install`
> 
> Now proceed as you would if `react-native init` was successful

It's working successfully this way, but when I create another project, I encounter the same error. Isn't there a permanent solution?

**Comment by takameyer** [comment]
The [fix](https://github.com/facebook/react-native/pull/39828) has been made, it just needs to be released.

**Comment by AdityaGoyal1999** [comment]
> # Workaround:
> When `react-native init` fails at cocoapods installation, do the following steps:
> 
> 1. Go to your project directory and open `Gemfile`
> 2. Replace Cocoapods version with `1.13`
> 3. Insert following line at the end of _Gemfile_ :  `gem 'activesupport', '~> 7.0.8'`
> 4. Run `bundle install` in your project directory
> 5. Go to `ios` directory in your project
> 6. Run `bundle exec pod install`
> 
> Now proceed as you would if `react-native init` was successful

I'm still facing an issue when I run `bundle install`

`In Gemfile:
  cocoapods was resolved to 1.13.0, which depends on
    cocoapods-core was resolved to 1.13.0, which depends on
      typhoeus was resolved to 1.4.0, which depends on
        ethon was resolved to 0.16.0, which depends on
          ffi
`

**Comment by cortinico** [comment]
Duplicates https://github.com/facebook/react-native/issues/39832
